drivers/onewire/onewire.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2016-2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_onewire
+ * @{
+ *
+ * @file
+ * @brief       1-Wire bus driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "xtimer.h"
+#include "onewire.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/* bus timings, all in microseconds */
+#define T_RESET_HOLD_US     (480)
+#define T_RESET_SAMPLE_US   (70)
+
+#define T_RW_PULSE_US       (3)
+#define T_W_HOLD_US         (57)
+#define T_W_RECOVER_US      (10)
+#define T_R_SAMPLE_US       (7)
+#define T_R_RECOVER_US      (55)
+
+static inline int _read_bit(const onewire_t *owi)
```

Comment:
Does this really have to be `static inline`?
The xtimer overhead will be much larger  than any function call overhead.

Reply:
nope, left over from back then where I did not know better :-)